### DISCUSSION TITLE
Is there an easy way to sync an s3 bucket to a local directory?

### DISCUSSION THREAD

**Discussion by lukehsiao** [dask/s3fs#671] [Q&A] [answered]
I have a directory of files stored in an s3 bucket, and want to be able to sync them to a local directory on my machine, edit them, and then push them back. Is there an easy way to do so with s3fs?

**Comment by lukehsiao** [comment]
Ah, I believe I've found a good example in the tests.

https://github.com/fsspec/s3fs/blob/f26f379c7198a65b6f3b001b415682d58b37c9ca/s3fs/tests/test_s3fs.py#L1857-L1866

**Comment by martindurant** [comment, marked as answer]
We don't explicitly have a rsync-like method right now. https://github.com/fsspec/filesystem_spec/issues/1074 discusses this - it would be an excellent new feature.
